project/src/main/java/com/google/autograder/data/Database.java

```diff
@@ -407,14 +406,49 @@ public String blobkeysFromAnswerGroupJson(String assignKey, String groupKey) {
     return blobKeyJson;
   }
 
-  //get all submissions for an assignment
-
-    // Get all answers for a question
-
-    // Set/edit location for a question
+  public static String getUngradedGroupKeys(String questionKey) {
+    Filter propertyFilter = new FilterPredicate("questionKey", FilterOperator.EQUAL, questionKey);
+    Query query = new Query("Group").setFilter(propertyFilter);
+    List<String> groupKeys = new ArrayList<String>();
+    for (Entity group : query(query)) {
+        if (group.getProperty("graded") != "GRADED") {
```

Comment:
This will need to be `!group.getProperty("graded").equals("GRADED")` (or `!"GRADED".equals(group.getProperty("graded"))` if you want to guard against null)